listenbrainz/webserver/views/profile.py

```diff
@@ -202,3 +212,33 @@ def upload():
 
         flash.info('Congratulations! Your listens from %d files have been uploaded successfully.' % success)
     return redirect(url_for("profile.import_data"))
+
+
+@profile_bp.route('/request-stats', methods=['GET'])
+@login_required
+def request_stats():
+    """ Check if the current user's statistics have been calculated and if not,
+        put them in the stats queue for stats_calculator.
+    """
+    status = _redis.redis.get(construct_stats_queue_key(current_user.musicbrainz_id)) == 'queued'
+    if status == 'queued':
+        flash.info('You have already been added to the stats calculation queue! Please check back later.')
+    elif db_stats.valid_stats_exist(current_user.id):
+        flash.info('Your stats were calculated in the most recent stats calculation interval,'
+            ' please wait until the next interval! We calculate new statistics every Monday at 00:00 UTC.')
```

Comment:
This is good, but out of date the second we change the cronjob. I think we should improve this in the future, but roll with it for now.